TrackingPerformance/Condor/condorJobs_sim.py

```diff
@@ -1,156 +1,248 @@
+
+            # Build ddsim command
+            arguments = [
+                # f" --compactFile /afs/cern.ch/work/g/gasadows/k4geo/FCCee/CLD/compact/{DetectorModelList_[0]}_3T/{DetectorModelList_[0]}.xml "
+                f" --compactFile $k4geo_DIR/{config.detModPaths['ILD_l5_v11']}",  # Note the change to use double quotes for the dictionary key
+                f"--outputFile {output_file_name}",
+                f"--steeringFile {config.sim_steering_file}",  # "CLDConfig/CLDConfig/cld_steer.py "
+                # TODO: FIX
+                # "--enableGun",
+                # f"--gun.particle {part}-",
+                # f"--gun.energy {momentum}*GeV",
+                # "--gun.distribution uniform",
+                # f"--gun.thetaMin {theta}*deg",
+                # f"--gun.thetaMax {theta}*deg",
+                # "--crossingAngleBoost 0",
+                # f"--numberOfEvents {Nevt_per_job}",
+            ]
+            command = f"ddsim {' '.join(arguments)} > /dev/null"
+
+            # Write bash script for job execution
+            bash_script = (
+                "#!/bin/bash \n"
+                f"source {environ_path} \n"
+                "git clone https://github.com/Victor-Schwan/TrackingStudies.git \n"  # FIXME: new repo location to config (to switch easily to local fork)
+                f"{command} \n"
+                f"xrdcp {output_file_name} root://eosuser.cern.ch/{output_dir} \n"
+            )
+            bash_file_name_parts = [
+                "bash_script",
+                dect,
+                part,
+                f"{theta}_deg",
+                f"{momentum}_GeV",
+                str(task_index),
+            ]
+            bash_file = (directory_jobs / "_".join(bash_file_name_parts)).with_suffix(
+                ".sh"
+            )
 
+            with open(bash_file, "w", encoding="utf-8") as b_file:
+                b_file.write(bash_script)
+                b_file.close()
+
+    if not need_to_create_scripts:
+        print("All output files are correct.")
+        sys.exit(0)
+
+    # ============================
+    # Condor Submission Script
+    # ============================
+    # Write the condor submission script
+    condor_script = (
+        "executable = $(filename) \n"
+        "arguments = $(ClusterId) $(ProcId) \n"
+        "output = output.$(ClusterId).$(ProcId).out \n"
+        "error = error.$(ClusterId).$(ProcId).err \n"
+        "log = log.$(ClusterId).log \n"
+        f'+JobFlavour = "{JOB_FLAVOR}" \n'
+        "queue filename matching files *.sh \n"
+    )
+    condor_file = directory_jobs / "condor_script.sub"
+    with open(condor_file, "w", encoding="utf-8") as c_file:
+        c_file.write(condor_script)
+        c_file.close()
+
+    # ====================
+    # Submit Job to Condor
+    # ====================
+    system(
+        "cd " + str(directory_jobs) + "; condor_submit condor_script.sub"
+    )  # FIXME: use subprocess instead?
```

Comment:
good idea :)
or maybe even get rid of this or make it optional so that one can inspect the file before submission.
I think we can expect a user to be able to still type this one line by hand. If we want to be extra nice we could even print a small message from which this is directly copy-pasteable.

Reply:
Or we could add a `no-submit` flag to `argparse`?

Reply:
yes a flag for this would also be a good idea but I am unsure if the default should be to submit it. :shrug:

Reply:
Another option is to outsource the execution and move it to a new script. This script can call the `main` function of `condorJobs_sim` and then submit the jobs.

Reply:
Moved to separate issue